app/uk/gov/hmrc/customs/declarations/information/connectors/ApiSubscriptionFieldsConnector.scala

```diff
@@ -32,25 +36,26 @@ class ApiSubscriptionFieldsConnector @Inject()(http: HttpClient,
                                                config: InformationConfigService)
                                               (implicit ec: ExecutionContext) {
 
-  def getSubscriptionFields[A](apiSubsKey: ApiSubscriptionKey)(implicit hci: HasConversationId, hc: HeaderCarrier): Future[ApiSubscriptionFieldsResponse] = {
-    val url = ApiSubscriptionFieldsPath.url(config.informationConfig.apiSubscriptionFieldsBaseUrl, apiSubsKey)
-    get(url)
-  }
-
-  private def get[A](url: String)(implicit hci: HasConversationId, hc: HeaderCarrier): Future[ApiSubscriptionFieldsResponse] = {
+  def getSubscriptionFields(apiSubsKey: ApiSubscriptionKey)
+                           (implicit hci: HasConversationId, hc: HeaderCarrier): Future[Option[ApiSubscriptionFieldsResponse]] = {
+    val url = new URL(ApiSubscriptionFieldsPath.url(config.informationConfig.apiSubscriptionFieldsBaseUrl, apiSubsKey))
     logger.debug(s"Getting fields id from api subscription fields service. url=$url")
 
-    http.GET[ApiSubscriptionFieldsResponse](url)
-      .recoverWith {
-        case upstreamErrorResponse: UpstreamErrorResponse =>
-          logger.error(s"Subscriptions fields lookup call failed. url=$url HttpStatus=${upstreamErrorResponse.statusCode} error=${upstreamErrorResponse.getMessage}")
-          Future.failed(upstreamErrorResponse)
-        case httpError: HttpException =>
-          logger.error(s"Subscriptions fields lookup call failed. url=$url HttpStatus=${httpError.responseCode} error=${httpError.getMessage}")
-          Future.failed(new RuntimeException(httpError))
-        case e: Throwable =>
-          logger.error(s"Call to subscription information service failed. url=$url")
-          Future.failed(e)
+    http.GET(url)
+      .map { response =>
+        response.status match {
+          case status if Status.isSuccessful(status) =>
+            Json.parse(response.body).asOpt[ApiSubscriptionFieldsResponse] match {
+              case Some(value) =>
+                Some(value)
+              case None =>
+                logger.error(s"Could not parse subscription fields response. url=$url")
+                None
+            }
+          case status =>
+            logger.error(s"Subscriptions fields lookup call failed. url=$url HttpStatus=$status")
```

Comment:
Could [] be put around the log variables?

Reply:
Yup, fixed.